numpyro/distributions/discrete.py

```diff
@@ -316,7 +316,9 @@ def log_prob(self, value):
         value = jnp.broadcast_to(value, batch_shape + (1,))
         logits = self.logits
         log_pmf = jnp.broadcast_to(logits, batch_shape + jnp.shape(logits)[-1:])
-        return jnp.take_along_axis(log_pmf, value, axis=-1)[..., 0]
+        return jnp.take_along_axis(log_pmf, jnp.array(value, dtype=int), axis=-1)[
```

Comment:
Why do we need this change? I thought `value` is already `int`?

Reply:
Oh, I see. One of the tests is failing. Could you update [this line](https://github.com/pyro-ppl/numpyro/blob/master/test/contrib/test_infer_discrete.py#L457) to `z in [0, 1]`?